### ISSUE TITLE
LH: Capture preconnect limitations

### ISSUE THREAD

**Issue by addyosmani** [GoogleChrome/lighthouse#10038]
**tl;dr Developers should include preconnects in the order they're to be utilized**

Chrome currently places a limit of 3 in-flight preconnects so developers don't need to compute how many hosts to preconnect to. This means developers connecting to a large number of hosts is WAI, but I don't believe most know the order of the preconnect origins is kinda important.

It may be worth Lighthouse include text in our docs to make it clear Chrome doesn't preconnect to all origins at the same time. So, developers should include preconnects in the order they are expected to be utilized.

Initially marking as a docs issue. We may want to factor this into the preconnect audit text if it doesn't get too wordy... :)

Confidence in this: confirmed with Tarun and Yoav
Source: https://cs.chromium.org/chromium/src/chrome/browser/predictors/preconnect_manager.h?rcl=226607d28a8085ffa3199f66954ee14c3e0736fb&l=138

Related fun tweet:
https://twitter.com/yoavweiss/status/1197551558071898113

**Comment by connorjclark** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=962693
> We should run a quick experiment to restrict preconnect to only origins that host render blocking content.

**Comment by paulirish** [comment]
related #9894 #9903

**Comment by paulirish** [comment]
and 2 year old discussion here https://github.com/GoogleChrome/lighthouse/issues/3106#issuecomment-333357398

**Comment by connorjclark** [comment]
I think we still need to tweak docs? https://github.com/GoogleChrome/lighthouse/pull/9903 fixed this is by emitting a warning when too many preconnects happen
